devito/finite_differences/differentiable.py

```diff
@@ -243,14 +243,16 @@ def laplace(self):
     def div(self, shift=None):
         space_dims = [d for d in self.dimensions if d.is_Space]
         return Add(*[getattr(self, 'd%s' % d.name)(x0=None if shift is None else
-                                                   d + shift * d.spacing)
-                     for d in space_dims])
+                                                   d + shift[i] * d.spacing if type(shift)
+                                                   is tuple else d + shift * d.spacing)
+                     for i, d in enumerate(space_dims)])
```

Comment:
Getting quite igly there too many if/if/....

Reply:
I agree with Mathias here. I think we cannot afford list comprehension here anymore.